spring-cloud-netflix-zuul/src/main/java/org/springframework/cloud/netflix/zuul/filters/SimpleRouteLocator.java

```diff
@@ -216,7 +216,7 @@ private String adjustPath(final String path) {
 		if (RequestUtils.isDispatcherServletRequest()
 				&& StringUtils.hasText(this.dispatcherServletPath)) {
 			if (!this.dispatcherServletPath.equals("/")
-					&& path.startsWith(this.dispatcherServletPath)) {
+					&& path.startsWith(this.dispatcherServletPath + "/")) {
```

Comment:
This never gets exercised by your new tests since `RequestUtils.isDispatcherServletRequest()` is always false.

Reply:
Adding
```
		RequestContext.getCurrentContext().set(IS_DISPATCHER_SERVLET_REQUEST_KEY, true);
// test
		RequestContext.getCurrentContext().remove(IS_DISPATCHER_SERVLET_REQUEST_KEY);
```
reveals that `test_getMatchingRouteFilterWithEqualServletPath()` fails with a null route

Reply:
What should be the expected behavior?
As I understand it, the purpose of this method is to remove the `dispatcherServletPath` from the path if someone puts it there. Isn't it better to just let it stay wrong, so the user can fix their bad configuration?

Reply:
a dispatcher path of `/bar` and a routepath of `/bar/**` will only match if the request path is `/bar/bar`. The tests need to be updated to add the request context setup I mentioned above and either add another test that matches a default route (ie dispatcher path of `/bar` and routepath of `/**`) or change the test that returns null to the values I described.

Reply:
I have changed the test to expect null